tastyworks/models/watchlists.py

```diff
@@ -39,45 +45,71 @@ def from_list(cls, data: list):
                     }
                 except KeyError:
                     inst.securities[item['symbol']] = {
-                        'instrument-type': None
+                        'instrument_type': None
                     }
         return inst
 
+    def __str__(self):
+        """String Representation"""
+        return str(self.__dict__)
+
 
 class WatchlistGroup(object):
-    """Watchlist Group
+    """WatchlistGroup Class"""
 
-    Generator for making document of watchlist objects.
+    def __init__(self):
+        """init"""
+        self.watchlists = {}
 
-    Notes:
-        Not all watchlists return group-name key's.  Hence the need to catch
-        an exception.
+    def __iter__(self):
+        """Iterator"""
+        return iter(self.watchlists)
 
-    Args:
-        session (TastyAPISession): Tastyworks API Session object
-        public (boolean): True if retrieving public-watchlists, false if
-                          retrieving custom-watchlists
+    def __getitem__(self, item):
+        """Subscriptor"""
+        return self.watchlists[item]
 
-    Returns:
-        WatchlistGroup: A group of watchlists
+    def __repr__(self):
+        """Object Representation"""
+        return str(self.watchlists)
 
-    """
-    def __init__(self):
-        self.watchlists = {}
+    def __str__(self):
+        """String Representation"""
+        return str(list(self.watchlists.keys()))
 
-    async def load_watchlists(
-        self, session: TastyAPISession, public: bool = True
+    @classmethod
+    async def get_watchlists(
+        cls, session: TastyAPISession, public: bool = True
     ):
+        """Get Watchlists
+
+        Class Factory
+
+        Retrieves a watchlist group (either public or private).
+
+        Notes:
+            Not all Watchlist's have group names.
+
+        Args:
+            session (TastyAPISession): A TastyAPISession object
+            public (bool): Retrive public or private watchlists
+
+        Returns:
+            WatchlistGroup: Instance of WatchlistGroup()
+        """
         url = f'{session.API_url}/public-watchlists' if public else f'{session.API_url}/watchlists'  # NOQA: E501
 
         async with aiohttp.request(
             'GET', url, headers=session.get_request_headers()
         ) as resp:
             if resp.status != 200:
-                raise Exception('Could not get public asset watchlists')
+                raise Exception(
+                    f'Status {resp.status}, message: {resp.json()["error"]["message"]}'  # NOQA: E501
```

Comment:
Would it make sense to prepend this message with some text stating what caused the exception? e.g. "Failed retrieving watchlist. Response status: ..."

Reply:
When it comes to making things more friendly, I usually won't back down.  Easy win/win for all here.